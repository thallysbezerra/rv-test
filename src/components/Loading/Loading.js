import React from 'react';
import './Loading.scss';

const Engine = () => (
	<div className="loading">
		<div className="loading__bar--1" />
		<div className="loading__bar--2" />
		<div className="loading__bar--3" />
	</div>
);

export default Engine;
