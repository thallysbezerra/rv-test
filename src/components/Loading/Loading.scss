@import '../../css/theme';

.loading {
	display: inline-block;
	height: 64px;
	left: 50%;
	position: absolute;
	top: 150px;
	transform: translateX(-50%);
	width: 64px;

	&__bar--1,
	&__bar--2,
	&__bar--3 {
		animation: loading 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;
		background-color: $red;
		display: inline-block;
		left: 6px;
		position: absolute;
		width: 13px;
	}

	&__bar--1 {
		animation-delay: -0.24s;
		left: 6px;
	}

	&__bar--2 {
		animation-delay: -0.12s;
		left: 26px;
	}

	&__bar--3 {
		animation-delay: 0;
		left: 45px;
	}
}
