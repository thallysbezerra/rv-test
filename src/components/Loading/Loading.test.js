import React from 'react';
import Loading from './Loading';

import renderer from 'react-test-renderer';

describe('Loading UI', () => {
	it('<Loading /> must have the snapshot version.', () => {
		const tree = renderer.create(<Loading />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
