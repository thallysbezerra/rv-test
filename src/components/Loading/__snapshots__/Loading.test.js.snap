// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading UI <Loading /> must have the snapshot version. 1`] = `
<div
  className="loading"
>
  <div
    className="loading__bar--1"
  />
  <div
    className="loading__bar--2"
  />
  <div
    className="loading__bar--3"
  />
</div>
`;
